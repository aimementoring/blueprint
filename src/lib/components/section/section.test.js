import React from 'react';
import renderer from 'react-test-renderer';
import Section from './section.js';

describe('Section', () => {
  it('renders properly without custom properties', () => {
    const tree = renderer.create(<Section />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly with grow value', () => {
    const tree = renderer.create(<Section grow />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly with justify content', () => {
    const tree = renderer.create(<Section justifyContent="flex-end" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly with align items', () => {
    const tree = renderer.create(<Section alignItems="center" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly with all 3 params', () => {
    const tree = renderer
      .create(<Section alignItems="start" justifyContent="flex-start" grow={2} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
