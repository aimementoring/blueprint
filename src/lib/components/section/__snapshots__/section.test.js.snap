// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders properly with align items 1`] = `
<div
  className="section"
  style={
    Object {
      "alignItems": "center",
    }
  }
/>
`;

exports[`Section renders properly with all 3 params 1`] = `
<div
  className="section"
  style={
    Object {
      "alignItems": "start",
      "flexGrow": 2,
      "justifyContent": "flex-start",
    }
  }
/>
`;

exports[`Section renders properly with grow value 1`] = `
<div
  className="section"
  style={
    Object {
      "flexGrow": true,
    }
  }
/>
`;

exports[`Section renders properly with justify content 1`] = `
<div
  className="section"
  style={
    Object {
      "justifyContent": "flex-end",
    }
  }
/>
`;

exports[`Section renders properly without custom properties 1`] = `
<div
  className="section"
  style={Object {}}
/>
`;
